test/mocks/mocks.go

```diff
@@ -360,8 +360,9 @@ func SecretMock(ns, name string) *corev1.Secret {
 			APIVersion: "v1",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ns,
-			Name:      name,
+			Namespace:       ns,
+			Name:            name,
+			ResourceVersion: "116076",
```

Comment:
I think we can not set or update resourceVersion it is done by cluster, this mock will work for the unit test but on actual cluster setting the resourceVersion will throw an error. The `get, list and watch operations` support the resourceVersion parameter. You might need to create another mock secret with the resourceVersion field for the unit test.

Reply:
Your e2e CI tests are failing for the same reason.
```
TestAddSchemesToFramework/end-to-end/scenario-sbr-db-app: servicebindingrequest_test.go:321: Creating Database credentials secret mock object...
    TestAddSchemesToFramework/end-to-end/scenario-sbr-db-app: servicebindingrequest_test.go:323: 
        	Error Trace:	servicebindingrequest_test.go:323
        	            				servicebindingrequest_test.go:540
        	            				servicebindingrequest_test.go:155
        	            				servicebindingrequest_test.go:103
        	Error:      	Received unexpected error:
        	            	resourceVersion should not be set on objects to be created
        	Test:       	TestAddSchemesToFramework/end-to-end/scenario-sbr-db-app
```